src/aws-cpp-sdk-core/include/aws/core/utils/crypto/PrecalculatedHash.h

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#pragma once
+
+#include <aws/core/Core_EXPORTS.h>
+#include <aws/core/utils/crypto/Hash.h>
+#include <aws/core/utils/crypto/HashResult.h>
+#include "aws/core/utils/Outcome.h"
+
+namespace Aws {
+    namespace Utils {
+        namespace Crypto {
+            /**
+             * CRC32 hash implementation.
+             */
+            class AWS_CORE_API PrecalculatedHash : public Hash {
+            public:
+                PrecalculatedHash(const Aws::String &mHash);
```

Comment:
nit: call it it hash instead of mHash lol. It really doesnt matter but we call member variables m_variable name to disambiguate from non-members